### PR TITLE
In start_registering, create register clients if reg list is empty

### PR DESCRIPTION
**Pull request by juha-h**
Also, had to change location of `create_register_clients` and `add_extension` functions.

### CODE REVIEW

File: src/ua.c [hunk 1]
```diff
@@ -138,6 +196,9 @@ static int start_register(struct ua *ua, bool fallback)
 	if (!fallback)
 		ua_event(ua, UA_EVENT_REGISTERING, NULL, NULL);
 
+	if (list_isempty(&ua->regl))
+		create_register_clients(ua);
+
```
Comment (cspiel1) [inline src/ua.c:198]: Does this change anything?

**Comment by juha-h** [comment]
Christian Spielberger writes:

> > @@ -138,6 +196,9 @@ static int start_register(struct ua *ua, bool fallback)
>  	if (!fallback)
>  		ua_event(ua, UA_EVENT_REGISTERING, NULL, NULL);
>  
> +	if (list_isempty(&ua->regl))
> +		create_register_clients(ua);
> +
> 
> Does this change anything?

Yes, it creates register clients if baresip was initially started with
regint=0.

**Comment by juha-h** [comment]
Just to add to my previous comment, register clients are not added by `create_register_clients` if regint=0 like it is when baresip is started with ;regint=0.  When /uareg is called with regint>0, subsequent call of `create_register_clients` does add them.

**Comment by cspiel1** [comment]
Okay, now I understand. So this should be okay.

**Comment by juha-h** [comment]
Christian Spielberger writes:

> Okay, now I understand. So this should be okay.

Thanks, I'll merge the PR.

**Pull request merged**